from PyQt5 import QtCore, QtGui, QtWidgets
from DataLogic import Data_Logic
from PyQt5 import QtTest


class Ui_RegisterWindow(object):
    def setupUi(self, RegisterWindow):
        RegisterWindow.setObjectName("RegisterWindow")
        RegisterWindow.resize(216, 252)
        palette = QtGui.QPalette()
        brush = QtGui.QBrush(QtGui.QColor(255, 255, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Base, brush)
        brush = QtGui.QBrush(QtGui.QColor(170, 170, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(255, 255, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Base, brush)
        brush = QtGui.QBrush(QtGui.QColor(170, 170, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(170, 170, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Base, brush)
        brush = QtGui.QBrush(QtGui.QColor(170, 170, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Window, brush)
        RegisterWindow.setPalette(palette)
        self.Label_FirstName = QtWidgets.QLabel(RegisterWindow)
        self.Label_FirstName.setGeometry(QtCore.QRect(10, 150, 71, 21))
        font = QtGui.QFont()
        font.setFamily("Tahoma")
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.Label_FirstName.setFont(font)
        self.Label_FirstName.setObjectName("Label_FirstName")
        self.Label_SignUp1 = QtWidgets.QLabel(RegisterWindow)
        self.Label_SignUp1.setGeometry(QtCore.QRect(80, 10, 51, 21))
        font = QtGui.QFont()
        font.setFamily("System")
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.Label_SignUp1.setFont(font)
        self.Label_SignUp1.setObjectName("Label_SignUp1")
        self.Label_SignUp2 = QtWidgets.QLabel(RegisterWindow)
        self.Label_SignUp2.setGeometry(QtCore.QRect(50, 30, 131, 21))
        font = QtGui.QFont()
        font.setFamily("System")
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.Label_SignUp2.setFont(font)
        self.Label_SignUp2.setObjectName("Label_SignUp2")
        self.Label_Username = QtWidgets.QLabel(RegisterWindow)
        self.Label_Username.setGeometry(QtCore.QRect(10, 90, 71, 21))
        font = QtGui.QFont()
        font.setFamily("Tahoma")
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.Label_Username.setFont(font)
        self.Label_Username.setObjectName("Label_Username")
        self.pushButton_Cancel = QtWidgets.QPushButton(RegisterWindow)
        self.pushButton_Cancel.setGeometry(QtCore.QRect(120, 210, 75, 23))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(9)
        font.setBold(True)
        font.setWeight(75)
        self.pushButton_Cancel.setFont(font)
        self.pushButton_Cancel.setAutoFillBackground(False)
        self.pushButton_Cancel.setObjectName("pushButton_Cancel")
        self.lineEdit_Password = QtWidgets.QLineEdit(RegisterWindow)
        self.lineEdit_Password.setGeometry(QtCore.QRect(90, 120, 113, 20))
        self.lineEdit_Password.setObjectName("lineEdit_Password")
        self.Label_LastName = QtWidgets.QLabel(RegisterWindow)
        self.Label_LastName.setGeometry(QtCore.QRect(10, 180, 71, 21))
        font = QtGui.QFont()
        font.setFamily("Tahoma")
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.Label_LastName.setFont(font)
        self.Label_LastName.setObjectName("Label_LastName")
        self.lineEdit_FirstName = QtWidgets.QLineEdit(RegisterWindow)
        self.lineEdit_FirstName.setGeometry(QtCore.QRect(90, 150, 113, 20))
        self.lineEdit_FirstName.setObjectName("lineEdit_FirstName")
        self.pushButton_Register = QtWidgets.QPushButton(RegisterWindow)
        self.pushButton_Register.setGeometry(QtCore.QRect(20, 210, 75, 23))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(9)
        font.setBold(True)
        font.setWeight(75)
        self.pushButton_Register.setFont(font)
        self.pushButton_Register.setObjectName("pushButton_Register")
        self.lineEdit_LastName = QtWidgets.QLineEdit(RegisterWindow)
        self.lineEdit_LastName.setGeometry(QtCore.QRect(90, 180, 113, 20))
        self.lineEdit_LastName.setObjectName("lineEdit_LastName")
        self.Label_Password = QtWidgets.QLabel(RegisterWindow)
        self.Label_Password.setGeometry(QtCore.QRect(10, 120, 71, 21))
        font = QtGui.QFont()
        font.setFamily("Tahoma")
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.Label_Password.setFont(font)
        self.Label_Password.setObjectName("Label_Password")
        self.lineEdit_Username = QtWidgets.QLineEdit(RegisterWindow)
        self.lineEdit_Username.setGeometry(QtCore.QRect(90, 90, 113, 20))
        self.lineEdit_Username.setObjectName("lineEdit_Username")
        self.Label_StatusSignUp = QtWidgets.QLabel(RegisterWindow)
        self.Label_StatusSignUp.setGeometry(QtCore.QRect(10, 60, 191, 21))
        font = QtGui.QFont()
        font.setFamily("System")
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.Label_StatusSignUp.setFont(font)
        self.Label_StatusSignUp.setObjectName("Label_StatusSignUp")
        
        self.pushButton_Register.clicked.connect(self.RegisterUser)
        self.pushButton_Cancel.clicked.connect(self.GoToLoginWin)

        self.retranslateUi(RegisterWindow)
        QtCore.QMetaObject.connectSlotsByName(RegisterWindow)

    def retranslateUi(self, RegisterWindow):
        _translate = QtCore.QCoreApplication.translate
        RegisterWindow.setWindowTitle(_translate("RegisterWindow", "Register"))
        self.Label_FirstName.setText(_translate("RegisterWindow", "First Name:"))
        self.Label_SignUp1.setText(_translate("RegisterWindow", "Sign Up"))
        self.Label_SignUp2.setText(_translate("RegisterWindow", "Create An Account"))
        self.Label_Username.setText(_translate("RegisterWindow", "Username:"))
        self.pushButton_Cancel.setText(_translate("RegisterWindow", "Cancel"))
        self.Label_LastName.setText(_translate("RegisterWindow", "Last Name:"))
        self.pushButton_Register.setText(_translate("RegisterWindow", "Register"))
        self.Label_Password.setText(_translate("RegisterWindow", "Password:"))
        self.Label_StatusSignUp.setText(_translate("RegisterWindow", "Status"))
        self.Label_StatusSignUp.hide()
        
    def RegisterUser(self):
        self.Label_StatusSignUp.show()
        
        Username = self.lineEdit_Username.text()
        Password = self.lineEdit_Password.text()
        FirstName = self.lineEdit_FirstName.text()
        LastName = self.lineEdit_LastName.text()
        
        if (Username == "" or Password == "" or FirstName == "" or LastName == ""):
            self.Label_StatusSignUp.setStyleSheet("color: red;")
            self.Label_StatusSignUp.setText("One Of The Fields Are Empty!")
        
        elif (Accounts.UserCheck(Username)):
            self.Label_StatusSignUp.setStyleSheet("color: red;")
            self.Label_StatusSignUp.setText("The Username Entered Already Exists!")
            
        else:
            Message = Accounts.AddUser(Username,Password,FirstName,LastName)
            self.Label_StatusSignUp.setStyleSheet("color: green;")
            self.Label_StatusSignUp.setText(Message)
                      
    def GoToLoginWin(self):
        username = 2
        
if __name__ == "__main__":
    import sys
    Accounts = Data_Logic()
    app = QtWidgets.QApplication(sys.argv)
    RegisterWindow = QtWidgets.QDialog()
    ui = Ui_RegisterWindow()
    ui.setupUi(RegisterWindow)
    RegisterWindow.show()
    sys.exit(app.exec_())

